Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -194,13 +195,25 @@ void ScrollTimeline::setTimelineScopeElement(const Element& element)
     m_timelineScopeElement = WeakPtr { &element };
 }
 
-ScrollableArea* ScrollTimeline::scrollableAreaForSourceRenderer(RenderElement* renderer, Ref<Document> document)
+ScrollableArea* ScrollTimeline::scrollableAreaForSourceRenderer(const RenderElement* renderer, const Document& document)
 {
     CheckedPtr renderBox = dynamicDowncast<RenderBox>(renderer);
     if (!renderBox)
         return nullptr;
 
-    if (renderer->element() == document->documentElement())
+    auto isRootScroller = [&] {
```

Comment:
The test is using `scrollingElement` (and runs in quirks mode because it has no DOCTYPE) so this code also needs to test `scrollingElement`.

Also add a non-quirks mode test maybe?

Reply:
There should also be a test where the body is independently scrollable from the root.